homeassistant/components/samsungtv/media_player.py

```diff
@@ -204,24 +215,18 @@ async def async_update(self) -> None:
         if not self._app_list_event.is_set():
             startup_tasks.append(self._async_startup_app_list())
 
-        if not self._upnp_device and self._ssdp_rendering_control_location:
-            startup_tasks.append(self._async_startup_upnp())
+        if not self._dmr_device and self._ssdp_rendering_control_location:
+            startup_tasks.append(self._async_startup_dmr())
 
         if startup_tasks:
             await asyncio.gather(*startup_tasks)
 
-        if not (service := self._get_upnp_service()):
+        if (dmr_device := self._dmr_device) is None:
             return
 
-        get_volume, get_mute = await asyncio.gather(
-            service.action("GetVolume").async_call(InstanceID=0, Channel="Master"),
-            service.action("GetMute").async_call(InstanceID=0, Channel="Master"),
-        )
-        LOGGER.debug("Upnp GetVolume on %s: %s", self._host, get_volume)
-        if (volume_level := get_volume.get("CurrentVolume")) is not None:
-            self._attr_volume_level = volume_level / 100
-        LOGGER.debug("Upnp GetMute on %s: %s", self._host, get_mute)
-        if (is_muted := get_mute.get("CurrentMute")) is not None:
+        if (volume_level := dmr_device.volume_level) is not None:
```

Comment:
Maybe break this out into another function so it can be called from the on_event?